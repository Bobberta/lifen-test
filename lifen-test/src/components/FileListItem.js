import React from 'react';

const FileListItem = (props) => {
  const { size, name } = props.file;
  const fileSize = Number(size / 1000).toFixed(1);
  return (
    <div className="file-list-item">
      <p className="file-list-item-name">{name}</p>
      <p>{`${fileSize}Ko`}</p>
    </div>
  );
};

export default FileListItem;
