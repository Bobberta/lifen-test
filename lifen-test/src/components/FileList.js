import React from 'react';
import FileListItem from './FileListItem';

const FileList = (props) => {
  const files = props.files || [];
  return (
    <div className="left-panel">
      <h3>Documents téléchargés</h3>
      {files.length === 0 ? (
        <div>Vous n'avez pas encore téléchargé de fichier.</div>
      ) : (
        files.map((file, index) => <FileListItem key={index} file={file} />)
      )}
    </div>
  );
};

export default FileList;
