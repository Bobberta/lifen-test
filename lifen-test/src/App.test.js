import { render, getByText } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import FileList from './components/FileList';
import FileListItem from './components/FileListItem';

const div = document.createElement('div');

const buffer = Buffer.from([1, 2, 3]);
const pdfFile = new File([buffer], 'Un fichier PDF', { type: 'application/pdf' });

describe('<FileList />', () => {
  it('renders without files', () => {
    const wrapper = shallow(<FileList files={[]} />);
    expect(wrapper.contains(<div>Vous n'avez pas encore téléchargé de fichier.</div>)).toEqual(
      true
    );
  });
  it('renders with files', () => {
    const wrapper = shallow(<FileList files={[pdfFile, pdfFile]} />);
    expect(wrapper.contains(<FileListItem key={0} file={pdfFile} />)).toEqual(true);
  });
  it('renders when files is undefined', () => {
    ReactDOM.render(<FileList />, div);
  });
});

describe('<FileListItem />', () => {
  it('renders correctly', () => {
    const wrapper = shallow(<FileListItem file={pdfFile} />);
    expect(wrapper).toMatchSnapshot();
  });
});
