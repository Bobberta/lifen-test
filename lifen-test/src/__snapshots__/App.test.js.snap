// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FileListItem /> renders correctly 1`] = `
<div
  className="file-list-item"
>
  <p
    className="file-list-item-name"
  >
    Un fichier PDF
  </p>
  <p>
    0.0Ko
  </p>
</div>
`;
